File: quick-library/src/main/java/org/quick/library/mvp/BaseView.java
package org.quick.library.mvp;

/**
 * Created by work on 2017/6/2.
 *
 * @author chris zou
 * @mail chrisSpringSmell@gmail.com
 */

public interface BaseView<T> {

    void setPresenter(T presenter);
    void showToast(String content);
    void showDialog(String content);
    void hideDialog();
}

File: quick-library/src/main/java/org/quick/library/mvp/BaseView.kt
package org.quick.library.mvp

import org.quick.component.QuickToast

/**
 * Created by work on 2017/6/2.
 *
 * @author chris zou
 * @mail chrisSpringSmell@gmail.com
 */

interface BaseView<T> {

    fun setPresenter(presenter: T)
    fun showToast(content: String) {
        QuickToast.showToastDefault(content)
    }

    fun showDialog(content: String)
    fun dismiss()
}
